aten/src/ATen/native/nested/NestedTensorMath.h

```diff
@@ -14,6 +14,122 @@ int64_t get_consistent_last_dim_of_nested_tensor(const NestedTensorImpl& nt);
 
 at::Tensor wrap_buffer(at::Tensor buffer, at::Tensor nested_size_tensor);
 
+at::Tensor wrap_buffer(at::Tensor buffer, at::Tensor nested_size_tensor, at::Tensor nested_stride_tensor, const std::vector<int64_t>& offsets);
+
+inline const at::Tensor& get_buffer(const at::Tensor& tensor) {
+  return get_nested_tensor_impl(tensor)->get_buffer();
+}
+
+// The starting positions of the underlying tensors in contiguous buffer memory
+// i.e. the buffer memory offsets to get the underlying tensors
+inline std::vector<int64_t> NestedTensor_get_offsets(const NestedTensorImpl* self_ptr) {
```

Comment:
hmm none of the functions in this file are new right? I don't see where they are moved from though in this PR

Reply:
might've been cleaner to separate out the refactor that moves things around without changing behavior into a separate PR in the stack

Reply:
This is true but also I am woking on the cluster and splitting up prs is a pain and if I didn't have to I would be super down with not splitting this PR

Reply:
The reason for the move though was that I added NestedTensorBackward function.. ohh wait you are right I needed them originally for the backward but I rebased and changed my impl of the backward formula so you are right I can move this back... that being said I still think it makes more sense having this inline functions live in the header file incase other files would like to use them.

Reply:
I see, keeping it here in the header is probably fine, but I don't think you removed it from the other file that you moved it from? I see 200 lines added, but softmax + shallow_copy_and_detach changes + testing only account for like 100ish.

Reply:
Ohhhh you are right, I rebased my changes on to what Yifen did and he renamed some of the functions. The Get_nested_shapes should no longer be needed let me remove those